go/ias/proxyclient/proxyclient.go

```diff
@@ -0,0 +1,140 @@
+// Package proxyclient implements the Oasis IAS proxy client endpoint.
+package proxyclient
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	tlsCert "github.com/oasislabs/oasis-core/go/common/crypto/tls"
+	cmnGrpc "github.com/oasislabs/oasis-core/go/common/grpc"
+	"github.com/oasislabs/oasis-core/go/common/identity"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/common/sgx/ias"
+	"github.com/oasislabs/oasis-core/go/ias/api"
+	"github.com/oasislabs/oasis-core/go/ias/proxy"
+)
+
+var _ api.Endpoint = (*proxyClient)(nil)
+
+type proxyClient struct {
+	identity *identity.Identity
+
+	conn     *grpc.ClientConn
+	endpoint api.Endpoint
+
+	spidInfo *api.SPIDInfo
+
+	logger *logging.Logger
+}
+
+func (c *proxyClient) fetchSPIDInfo(ctx context.Context) error {
+	if c.spidInfo != nil || c.endpoint == nil {
+		return nil
+	}
+
+	var err error
+	if c.spidInfo, err = c.endpoint.GetSPIDInfo(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *proxyClient) VerifyEvidence(ctx context.Context, evidence *api.Evidence) (*ias.AVRBundle, error) {
```

Comment:
Please provide comment since this is public.

Reply:
This is not public as `proxyClient` is a private type.